docs/configuration/configuration.md

```diff
@@ -275,13 +275,17 @@ See below for the configuration options for Azure discovery:
 # The information to access the Azure API.
 # The Azure environment.
 [ environment: <string> | default = AzurePublicCloud ]
-# The subscription ID.
+
+# The authentication method, either OAuth or ManagedIdentity.
+# See https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview
+[ authentication_method: <string> | default = OAuth]
+# The subscription ID. Always required.
 subscription_id: <string>
-# The tenant ID.
+# The tenant ID. Only required with authentication_method OAuth.
 tenant_id: <string>
```

Comment:
This should be marked optional. And below.

Reply:
They are kinda semi-optional, as they are only optional when using ManagedIdentity (in which case they are not even used).

However, they are required when using OAuth as authentication method (which is the default) and Prometheus will fail hard when they are not set in this case (introduced with #4907 )

Reply:
They're only required if they're always required.

Reply:
OK, I have marked them now as optional in the docs.